src/commands/sourcemaps/explain.rs

```diff
@@ -0,0 +1,404 @@
+        .alias("why")
+        .arg(
+            Arg::new("event")
+                .value_name("EVENT_ID")
+                .required(true)
+                .help("ID of an event to be explained."),
+        )
+        .arg(
+            Arg::new("force")
+                .long("force")
+                .short('f')
+                .help("Force full validation flow, even when event is already source mapped."),
+        )
+}
+
+fn tip<S>(msg: S)
+where
+    S: std::fmt::Display,
+{
+    println!("{}", style(format!("ℹ {}", msg)).blue());
+}
+
+fn success<S>(msg: S)
+where
+    S: std::fmt::Display,
+{
+    println!("{}", style(format!("✔ {}", msg)).green());
+}
+
+fn warning<S>(msg: S)
+where
+    S: std::fmt::Display,
+{
+    println!("{}", style(format!("⚠ {}", msg)).yellow());
+}
+
+fn error<S>(msg: S)
+where
+    S: std::fmt::Display,
+{
+    println!("{}", style(format!("✖ {}", msg)).red());
+}
+
+fn fetch_event(org: &str, project: &str, event_id: &str) -> Result<ProcessedEvent<'static>> {
+    match Api::current().get_event(org, Some(project), event_id)? {
+        Some(event) => {
+            success(format!("Fetched data for event: {}", event_id));
+            Ok(event)
+        }
+        None => {
+            error(format!("Could not retrieve event {}", event_id));
+            tip("Make sure that event ID you used is valid.");
+            Err(QuietExit(1).into())
+        }
+    }
+}
+
+fn extract_top_frame(stacktrace: &Stacktrace) -> Result<&Frame> {
+    let in_app_frames: Vec<&Frame> = stacktrace
+        .frames
+        .iter()
+        .filter(|frame| frame.in_app.unwrap_or(false))
+        .collect();
+
+    if in_app_frames.is_empty() {
+        bail!("Event exception stacktrace has no in_app frames");
+    }
+
+    let top_frame = in_app_frames.last().unwrap();
+    let abs_path = top_frame
+        .abs_path
+        .as_ref()
+        .ok_or_else(|| format_err!("Top frame is missing an abs_path"))?;
+
+    if let Ok(abs_path) = Url::parse(abs_path) {
+        if Path::new(abs_path.path()).extension().is_none() {
+            bail!("Top frame of event exception originates from the <script> tag, its not possible to resolve source maps");
+        }
+    } else {
+        bail!("Event exception stacktrace top frame has incorrect abs_path (valid url is required). Found {}", abs_path);
+    }
+
+    Ok(top_frame)
+}
+
+fn fetch_release_artifacts(org: &str, project: &str, release: &str) -> Result<Vec<Artifact>> {
+    Api::current().list_release_files(org, Some(project), release).map(|artifacts| {
+        if artifacts.is_empty() {
+            error("Release has no artifacts uploaded");
+            tip("https://docs.sentry.io/platforms/javascript/sourcemaps/troubleshooting_js/#verify-artifacts-are-uploaded");
+            return Err(QuietExit(1).into());
+        }
+        Ok(artifacts)
+    })?
+}
+
+fn find_matching_artifact(artifacts: &[Artifact], abs_path: &str) -> Result<Artifact> {
```

Comment:
this is the only function here I found really hard to follow because I don't know what `abs_path` usually looks like and why we add a `~` later. Maybe add a comment with an example?